File: internal/judge/judge.go
package judge

import (
	"os"
	"strings"

	"judge/internal/sandbox"
)

func JudgeSubmission(code, language, questionID string) (string, string) {
	inputPath := "Questions/" + questionID + "_in.txt"
	outputPath := "Questions/" + questionID + "_out.txt"

	inputData, err := os.ReadFile(inputPath)
	if err != nil {
		return "SYSTEM_ERROR", err.Error()
	}

	expectedOutput, err := os.ReadFile(outputPath)
	if err != nil {
		return "SYSTEM_ERROR", err.Error()
	}

	out, runtimeErr := sandbox.RunCode(
		code,
		language,
		string(inputData),
	)

	if runtimeErr == "TLE" {
		return "TLE", ""
	}

	if runtimeErr != "" {
		return "RUNTIME_ERROR", runtimeErr
	}

	if strings.TrimSpace(out) != strings.TrimSpace(string(expectedOutput)) {
		return "WRONG_ANSWER", "Output mismatch"
	}

	return "ACCEPTED", "All test cases passed"
}

File: pythonMirror/judge.py
from sandbox import run_code

def judge_submission(code, language, question_id):
    input_path = f"Questions/{question_id}_in.txt"
    output_path = f"Questions/{question_id}_out.txt"

    with open(input_path) as f:
        inp = f.read()

    with open(output_path) as f:
        expected = f.read()

    out, err = run_code(
        code,
        language,
        inp
    )

    if err == "TLE":
        return "TLE", ""

    if err:
        return "RUNTIME_ERROR", err

    if out.strip() != expected.strip():
        return (
            "WRONG_ANSWER",
            f"Expected {expected}, got {out}"
        )

    return "ACCEPTED", "All test cases passed"
